zapcore/write_syncer_bench_test.go

```diff
@@ -53,4 +56,38 @@ func BenchmarkMultiWriteSyncer(b *testing.B) {
 			}
 		})
 	})
+	b.Run("4 discarder with buffer", func(b *testing.B) {
+		w := &BufferedWriteSyncer{
+			WriteSyncer: NewMultiWriteSyncer(
+				&ztest.Discarder{},
+				&ztest.Discarder{},
+				&ztest.Discarder{},
+				&ztest.Discarder{},
+			),
+		}
+		defer w.Close()
+		b.ResetTimer()
+		b.RunParallel(func(pb *testing.PB) {
+			for pb.Next() {
+				w.Write([]byte("foobarbazbabble"))
+			}
+		})
+	})
+}
+
+func BenchmarkWriteSyncer(b *testing.B) {
+	b.Run("write file with no buffer", func(b *testing.B) {
+		file, err := ioutil.TempFile("", "log")
+		assert.NoError(b, err)
```

Comment:
any reason for catching errors before defering the file removal? Do you even need to manually Close or does os.Remove do that for you?